doc/docs/master/deploy-manage/deploy/envoy.md

```diff
@@ -0,0 +1,487 @@
+
+!!! Note "TL;DR" 
+    - When Envoy is activated, Pachyderm is reachable through **one TCP port for all incoming grpc (grpcs), console (HTTP/HTTPS), s3 gateway, OIDC, and dex traffic**, then routes each call to the appropriate backend microservice without any additional configuration.
+    - To deploy Pachyderm with Envoy, enable the proxy as follow:
+
+    ```yaml
+    proxy:
+      enabled: true
+      service:
+        type: LoadBalancer
+    ```
+
+!!! Warning
+    Envoy is an optional deployment option that will become permanent in the next minor release of Pachyderm.
+
+The high-level architecture diagram below gives a quick overview of the layout of services and pods when using Envoy. In particular, it details how Pachyderm listens to all inbound traffic on one port, then routes each call to the appropriate backend:![Infrastruture Recommendation](../../images/infra-recommendations-envoy.png)
+
+!!! Note 
+    See our [reference values.yaml](https://github.com/pachyderm/pachyderm/blob/master/etc/helm/pachyderm/values.yaml#L649){target=_blank} for all available configurable fields of the proxy.
+
+Before any deployment in production, we recommend reading the following recommendations section and [set up your production infrastructure](#deploy-pachyderm-in-production-with-envoy). 
+
+Alternatively, you can skip those infrastructure prerequisites and make a [quick cloud installation](#quick-cloud-deployment-with-envoy) or jump to our [local deployment](#deploy-pachyderm-locally-with-envoy) section for the first encounter with Pachyderm.
+
+## Pachyderm General Infrastructure Recommendations
+
+For production deployments, we recommend that you:
+
+* **Provision a TCP load balancer** for all HTTP/HTTPS, gRPC/gRPCs, aws s3, /dex incoming traffic.
+The TCP load balancer (load balanced at L4 of the OSI model) will have port `80/443` forwarding to the `pachyderm-proxy` service entry point. Please take a look at the diagram above.
+        
+    When Envoy is enabled with `type:LoadBalancer` (see the snippet of values.yaml enabling the proxy), Pachyderm creates a `pachyderm-proxy` service allowing your cloud platform (AWS, GKE...) to **provision a TCP Load Balancer automatically**.
+        
+    To provision this external load balancer automatically (if supported) and attach any Load Balancer configuration information to the metadata of your servic, add the appropriate `annotations` in the `proxy.service` of your values.yaml (see below):
+
+    ```yaml
+    proxy:
+      enabled: true
+      service:
+        type: LoadBalancer
+        annotations: {see examples below}
+    ```
+
+
+    === "Example on AWS EKS"
+        In the following example, we deploy an NLB on AWS EKS:
```

Comment:
Why are we choosing an NLB here?  I find that they work very poorly with k8s; a "classic" load balancer is fine though.

Reply:
removed all annotations